types/github-username-regex/index.d.ts

```diff
@@ -0,0 +1,7 @@
+// Type definitions for github-username-regex 1.0
+// Project: https://github.com/shinnn/github-username-regex
+// Definitions by: BehindTheMath <https://github.com/BehindTheMath>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare const githubUsernameRegex: RegExp;
+export default githubUsernameRegex;
```

Comment:
It looks like this should actually be `export =`? The [source](https://github.com/shinnn/github-username-regex/blob/master/index.js) code uses `module.exports =` and not `export default`.

Reply:
That's a good point. However, the [source code](https://github.com/shinnn/github-username-regex/blob/0794566cc10e8c5a0e562823f8f8e99fa044e5f4/module.js) also has parallel code for ES6 modules.

I can't find any clear directives, but it seems from the [TS docs on modules](https://www.typescriptlang.org/docs/handbook/modules.html) and [this issue](https://github.com/Microsoft/TypeScript/issues/2242) that going forward, the recommended format is to use ES6 imports, and `export =` is supported just for backwards compatibility.

So the recommended way of importing it in Typescript would be (as shown in the test file):
```typescript
import githubUsernameRegex from "github-username-regex";
```
When TSC does type-checking, it will use this type definition, which matches the export from `module.js`.
If it's compiled to ES6 modules (`"module": "es6"`), the import statement will remain the same.
If it's compiled to CommonJS modules (`"module": "commonjs"`), it will be compiled as:
```typescript
const githubUsernameRegex = require("github-username-regex");
```
and it will be resolved from `index.js`.

**To simplify**: The current type definition matches the ES6 implementation of the library, and I believe that typings for the CommonJS implementation are not necessary.

Reply:
The typings should describe the library as it's imported the usual way, which in node is with `require()`. A default import compiles to `require("github-username-regex").default`, which is wrong since `require("github-username-regex")` is a regexp which has no `default` property.
You may be interested in Microsoft/TypeScript#16093 which would allow `export =` to be imported with a default import.

Reply:
My mistake; you are correct.

The question then becomes flipped. Once the definitions are changed to `export =`, what happens if I want to output ES6 modules (`"module": "es6"`)?

Reply:
For now you can use `--allowSyntheticDefaultImports` which will let you use default imports even for modules with `export =`.